### PR TITLE
[PVR] CPVRClient: Refactor writing addon stream properties to file item.

### PR DESCRIPTION
**Pull request by ksooo**
Just a small code refactoring. No functionality should be changed by this.

This change has been runtime-tested on macOS, latest Kodi master.

@peak3d this effects and simplifies #12755 

@Jalle19 mind taking a look at the code change.

### CODE REVIEW

**Comment by peak3d** [comment]
@ksooo usuful refactoring :-) But how do I set mimetype now?

**Comment by ksooo** [comment]
Not sure I understand what you mean. Just do "fileItem.SetFoo" in the newly added function, like you did before.

File: xbmc/addons/PVRClient.cpp [hunk 1]
```diff
@@ -808,6 +808,17 @@ PVR_ERROR CPVRClient::IsPlayable(const CConstPVREpgInfoTagPtr &tag, bool &bIsPla
   return retVal;
 }
 
+void CPVRClient::WriteFileItemProperties(const PVR_NAMED_VALUE *properties, unsigned int iPropertyCount, CFileItem &fileItem)
```
Comment (xhaggi) [inline xbmc/addons/PVRClient.cpp]: Minor but `Set` fits better as `Write`

Comment (xhaggi) [reply xbmc/addons/PVRClient.cpp]: And put fileitem param as first?

File: xbmc/addons/PVRClient.h [hunk 2]
```diff
@@ -905,6 +905,11 @@ namespace PVR
     bool LogError(const PVR_ERROR error, const char *strMethod) const;
 
     /*!
+     * @brief Write the given addon properties into given file item's properties.
```
Comment (xhaggi) [inline xbmc/addons/PVRClient.h]: Would be good to add docs for params too.

File: xbmc/addons/PVRClient.cpp (hunk 1 above)
Comment (ksooo) [reply xbmc/addons/PVRClient.cpp]: Not really, as all CPVRClient methods copying data back and forth between add-on and Kodi are prefixed "Write". I actually started with "Set" and then changed my code  to fit  the wording already used here.

Comment (xhaggi) [reply xbmc/addons/PVRClient.cpp]: Ok don't know that

Comment (ksooo) [reply xbmc/addons/PVRClient.cpp]: Re second comment: no, as I ffollow source target order.

File: xbmc/addons/PVRClient.h (hunk 2 above)
Comment (ksooo) [reply xbmc/addons/PVRClient.h]: I think the @brief explains it good enough, adding documentation for the parameters is no real value add here. And it is a private method anyway.

Comment (xhaggi) [reply xbmc/addons/PVRClient.h]: Ok I'll shut up now 😄

Comment (ksooo) [reply xbmc/addons/PVRClient.h]: Thanks anyway.

**Review by Jalle19 — APPROVED** [review]

**Pull request merged**